landa/patches/set_phone_and_email_as_primary.py

```diff
@@ -0,0 +1,46 @@
+import frappe
+
+
+def execute():
+    def value_not_set(dict, field):
+        return len(dict) > 0 and (1 not in [d[field] for d in dict])
+
+    contacts = frappe.get_list("Contact", as_list=True)
+    for contact in contacts:
+        # if no email_id is set as primary for a contact, set the first email_id as primary
```

Comment:
As a rule of thumb, in quite a few cases when you feel like writing a comment, you should actually implement a function instead (in this case, for example, `set_primary_email_if_missing(contact)` ). This keeps the functions short, the source code clean and self-explanatory.

Reply:
Thanks, I tried to refactor accordingly.